astropy/io/misc/pickle_helpers.py

```diff
@@ -4,12 +4,23 @@
 part of a larger framework or standard.
 """
 
+import warnings
 
+from ...utils.exceptions import AstropyDeprecationWarning
 
 __all__ = ['fnpickle', 'fnunpickle']
 
 
-def fnunpickle(fileorname, number=0, usecPickle=True):
+def fnunpickle(filename, **kwargs):
```

Comment:
I don't like the change of signature. It makes `number` keyword-only. Why not do something like this instead:

```
class _Default:
    # with custom representation so the sphinx build shows a "nice" signature
    def __repr__(self):  
        return 'default'

_default = _Default()

def fnunpickle(filename, number=0, usecPickle=_default):
    if usecPickle is not _default:
       ...
```

Reply:
Thanks, indeed this is  good idea. At the end I've followed numpy's NoValue approach.